File: devel/test-triangle.cxx
#include <cstdio>
#include <iostream>
#include <ctime>
#include <cmath>
#include "numcxx/numcxx.hxx"



int main(void)
{
    auto pGeometry=numcxx::Geometry::create();
    pGeometry->set_points({
            {0,0},
            {1,0},
            {1,1}});
    pGeometry->set_bfaces({
            {0,1},
            {1,2},
            {2,0}});
    pGeometry->set_bfaceregions({0,1,2});

    pGeometry->set_regionpoints({{0.5,0.5}});
    pGeometry->set_regionnumbers({1});
    pGeometry->set_regionvolumes({0.01});

    
        
    auto pGrid=numcxx::SimpleGrid::create(pGeometry,"zpaAqV");
    std::cout<< *pGrid->bfaceregions;
}

File: examples/part4/test-triangle.py
#!/usr/bin/env python
import numcxx
import numpy
import matplotlib
from matplotlib import pyplot as plt



geom=numcxx.Geometry.create()
geom.set_points([[0,0.1],[1,0],[1.1,1]])
geom.set_bfaces([[0,1],[1,2],[2,0]])
geom.set_bfaceregions([1,2,3])
geom.set_regionpoints([[0.5,0.25]])
geom.set_regionvolumes([0.001])
geom.set_regionnumbers([1])

print(numcxx.asnumpy(geom.bfaceregions))



numcxx.plot.plotGeometry(plt,geom)
plt.show()
grid=numcxx.SimpleGrid.create(geom,"zpaAq")










numcxx.plot.plotGrid(plt,grid)
plt.show()
print(matplotlib.get_backend())
